### PR TITLE
Enable the documentation URL in the default menu items to be configur…

### PR DESCRIPTION
**Pull request by objectiser**
…ed via the operator CLI

Signed-off-by: Gary Brown <gary@brownuk.com>

### CODE REVIEW

File: pkg/cmd/start/main.go [hunk 1]
```diff
@@ -89,6 +89,9 @@ func NewStartCommand() *cobra.Command {
 	cmd.Flags().Int32("cr-metrics-port", 8686, "The metrics port for Operator and/or Custom Resource based metrics")
 	viper.BindPFlag("cr-metrics-port", cmd.Flags().Lookup("cr-metrics-port"))
 
+	cmd.Flags().String("documentation-url", "https://www.jaegertracing.io/docs/latest", "The URL for the 'Documentation' menu item")
```
Comment (objectiser) [inline pkg/cmd/start/main.go]: Not sure whether to make the default URL use the version, rather than latest?

Comment (jpkrohling) [reply pkg/cmd/start/main.go]: I think it's a great idea to tie it to the version.

File: pkg/strategy/controller.go [hunk 2]
```diff
@@ -231,13 +231,18 @@ func enableLogOut(uiOpts map[string]interface{}, spec *v1.JaegerSpec) {
 		return
 	}
 
+	docURL := viper.GetString("documentation-url")
+	if docURL == "" {
+		docURL = "https://www.jaegertracing.io/docs/latest"
```
Comment (jpkrohling) [inline pkg/strategy/controller.go]: This shouldn't be necessary, as you should get the default value from viper.

File: pkg/strategy/controller.go [hunk 3]
```diff
@@ -231,13 +231,18 @@ func enableLogOut(uiOpts map[string]interface{}, spec *v1.JaegerSpec) {
...
+	}
+
 	menuStr := `[
 		{
 		  "label": "About",
 		  "items": [
 			{
 			  "label": "Documentation",
-			  "url": "https://www.jaegertracing.io/docs/latest"
+			  "url": "` + docURL + `"
```
Comment (jpkrohling) [inline pkg/strategy/controller.go]: Might be cleaner to use `%s` here and use `menuStr = fmt.Sprintf(menuStr, docURL)` later.

File: pkg/strategy/controller.go (hunk 2 above)
Comment (objectiser) [reply pkg/strategy/controller.go]: Yeah thought that - but when running the unit test from within vscode it failed.

File: pkg/strategy/controller.go (hunk 3 above)
Comment (objectiser) [reply pkg/strategy/controller.go]: ok no problem

File: pkg/strategy/controller.go (hunk 2 above)
Comment (jpkrohling) [reply pkg/strategy/controller.go]: The default value is set in the command, so, it's not being set at all during the unit tests. You'll need something similar to this:

https://github.com/jaegertracing/jaeger-operator/blob/e377c2bc3a5008bd573ad177ff5f6036af2f3649/pkg/deployment/agent_test.go#L17-L23

Comment (objectiser) [reply pkg/strategy/controller.go]: Ok thanks.

**Pull request merged**